targets/UnrealMarketplacePlugin/makeCpp.js

```diff
@@ -611,10 +637,17 @@ function getRequestActions(tabbing, apiCall) {
 function getResultActions(tabbing, apiCall) {
     if (apiCall.url === "/Authentication/GetEntityToken")
         return tabbing + "if (outResult.EntityToken.Len() > 0)\n"
-            + tabbing + "    PlayFabSettings::SetEntityToken(outResult.EntityToken);\n\n";
+            + tabbing + "    PlayFabSettings::SetEntityToken(outResult.EntityToken);\n";
     else if (apiCall.result === "LoginResult")
-        return tabbing + "if (outResult.SessionTicket.Len() > 0) PlayFabSettings::SetClientSessionTicket(outResult.SessionTicket);\n"
-            + tabbing + "if (outResult.EntityToken.IsValid()) PlayFabSettings::SetEntityToken(outResult.EntityToken->EntityToken);\n"
+        return tabbing + "outResult.AuthenticationContext = TSharedPtr<UPlayFabAuthenticationContext>(NewObject<UPlayFabAuthenticationContext>());\n"
```

Comment:
A UObject can be garbage-collected if it's used with UPROPERTY fields.  TSharedPtr is refcounted and is not used with UPROPERTY.  The same class can coexist with both systems - it's a matter of how they're consumed and instantiated.  For the C++ API, we had a choice between
A) Use TSharedPtr
B) Make every struct that contains `AuthenticationContext` a USTRUCT, and all fields that store them UPROPERTY.

B was a more significant change, and requires customers to refactor any existing code that stores `PlayFabCppBaseModel` implementations (if they have existing code that stores these structs as non-UPROPERTY).

Given this information, please let me know if you think that option B would still be better, such as for the sake of consistency.

Reply:
Making a `TSharedPtr<UClass>` is perfectly safe as long as `RF_Standalone` is set, which prevents the garbage collector from destroying the object (until shutdown).  `~UObject()` removes itself from the object table, so it's safe that `TSharedPtr` dereference will `delete` it.

Just in case, I just did some additional testing and research to try to identify a reason this could be dangerous, and couldn't find any.